File: python3/1844.py
class Solution:
    def replaceDigits(self, s: str) -> str:
        n = len(s)
        r = []
        x = "abcdefghijklmnopqrstuvwxyz"
        for i in range(0, n, 2):
            r.append(s[i])
            if i + 1 < n:
                r.append(x[ord(s[i]) + ord(s[i + 1]) - 145])
        return "".join(r)

File: dart/1844.dart
class Solution {
    String replaceDigits(String s) {
        int n = s.length;
        String r = "", x = "abcdefghijklmnopqrstuvwxyz", y = "0123456789";
        for (int i = 0; i < n; i += 2) {
            r += s[i];
            if (i + 1 < n)
                r += x[x.indexOf(s[i]) + y.indexOf(s[i + 1])];
        }
        return r;
    }
}
